File: mysql/csharp/read/mysql_read.cs
// ----------------------------------------------------------------
/*
	read/mysql_read.cs

					Sep/17/2013

*/
// ----------------------------------------------------------------
using	System;
using	System.Data;
using	MySql.Data.MySqlClient;

// ----------------------------------------------------------------
public static class mysql_read
{
public static void Main (string[] args)
{
	Console.WriteLine ("*** 開始 ***");

	string server="host_mysql";
	string str_db="city";

	string str_connect = "Server=" + server + 
	";User Id=scott;Password=tiger;" +
		"Database=" + str_db + ";";

	MySqlConnection connection = new MySqlConnection (str_connect);
	connection.Open ();

	DataTable dtable = mysql_manipulate.mysql_data_fetch_proc (connection);

	connection.Close ();

	table_manipulate.display_proc (dtable);

	Console.WriteLine("*** 終了 ***");
}

// ----------------------------------------------------------------
}
// ----------------------------------------------------------------

File: mysql/fsharp/read/mysql_read.fs
// -------------------------------------------------------------------
//
//	mysql_read.fs
//
//					Jul/12/2011
//
// -------------------------------------------------------------------
open System
open System.Data
open MySql.Data.MySqlClient

// -------------------------------------------------------------------
[<EntryPoint>]
let main (args : string[] ) =
  printfn "*** 開始 ***"

  let conn = new MySqlConnection "userid=scott;password=tiger;database=city;Host=host_mysql"
  conn.Open()

  let cmd = new MySqlCommand( "SELECT * FROM cities", conn )
  let reader = cmd.ExecuteReader()

  while reader.Read()
    do System.Console.Write (reader.GetString 0 )
       System.Console.Write ("\t")
       System.Console.Write (reader.GetString 1 )
       System.Console.Write ("\t")
       System.Console.Write (reader.GetInt32 2 )
       System.Console.Write ("\t")
       System.Console.WriteLine (reader.GetDateTime 3 )
       done

  reader.Close()
  conn.Close()

  printfn "*** 終了 ***"

  0

// -------------------------------------------------------------------
